assets/scss/components/_product-compare.scss

```diff
@@ -1,5 +1,25 @@
 $details-border: 1px solid $color-neutrals-75;
 
+.c-products-header {
+    display: block;
+
+    .no-results & {
+        display: none;
+    }
+
+    &--empty {
```

Comment:
It's more a suggestion, but instead hading two h1 tags, you cold have two spans inside h1 tag and toggle spans visibility instead of having more complicated code like this :D